plugins/inputs/etherscan/etherscan.go

```diff
@@ -0,0 +1,244 @@
+package etherscan
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/models"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type EtherscanOption func(*Etherscan)
+
+func WithHTTPClient(newClient HTTPClient) EtherscanOption {
+	return func(plugin *Etherscan) {
+		plugin.client = newClient
+	}
+}
+
+func WithLogger(newLogger telegraf.Logger) EtherscanOption {
+	return func(plugin *Etherscan) {
+		plugin.log = newLogger
+	}
+}
+
+type Etherscan struct {
+	log telegraf.Logger
+
+	client          HTTPClient
+	Timeout         string `toml:"req_timeout_duration"`
+	timeoutDuration time.Duration
+
+	networkRequests requestQueue
+	Key             string `toml:"api_key" required:"true"`
+	requestCount    int
+
+	RequestLimit            int    `toml:"api_rate_limit_calls"`
+	RequestInterval         string `toml:"api_rate_limit_duration"`
+	requestIntervalDuration time.Duration
+	burstLimiter            chan time.Time
+
+	Balances []balance `toml:"balance"`
+
+	initializeOnce sync.Once
+}
+
+func NewEtherscanInputPlugin(opts ...EtherscanOption) *Etherscan {
+	nep := Etherscan{
+		client:          &http.Client{},
+		log:             models.NewLogger("input", "etherscan", ""),
+		networkRequests: make(requestQueue),
+		Key:             "",
+		Timeout:         _RequestTimeout,
+		RequestLimit:    _RequestLimit,
+		RequestInterval: _RequestInterval,
+		Balances:        make([]balance, 0),
+		initializeOnce:  sync.Once{},
+	}
+
+	for _, opt := range opts {
+		opt(&nep)
+	}
+
+	return &nep
+}
+
+type _ telegraf.Input
+
+func (m *Etherscan) Description() string {
+	return _Description
+}
+
+func (m *Etherscan) SampleConfig() string {
+	return _ExampleEtherscanConfig
+}
+
+func (m *Etherscan) Setup() error {
+	reqInvDur, err := time.ParseDuration(m.RequestInterval)
+	if err != nil {
+		return err
+	}
+	m.requestIntervalDuration = reqInvDur
+
+	reqTimeoutDur, err := time.ParseDuration(m.Timeout)
+	if err != nil {
+		return err
+	}
+	m.timeoutDuration = reqTimeoutDur
```

Comment:
Telegraf can do this for you if you define those variables as `internal.Duration`.